### PR TITLE
Add new parameter control Windows Performance Counter refresh rate

### PR DESCRIPTION
**Pull request by steveny91**
### What does this PR do?
Adds a new parameter on the agent level that allows for windows performance counters refresh periods to be controlled. Refreshes are necessary to discover new counter/counterset and removal of old instances.

### Motivation

In some cases it's observed that possibly due to permissions issues, windows events are generated when the counters are refreshed. This is a side effect and is not known to interfere with the check itself. This param will allow customer to increase the interval should they feel that events getting generated at 1 minute intervals is still too much. 

This has been reported in a few support cases.

### Additional Notes
Linked PR:
https://github.com/DataDog/integrations-core/pull/12665

### Possible Drawbacks / Trade-offs
Less up to date data when new counters are created/deleted. The refresh is crucial when there's new counters are exposed.

### Describe how to test/QA your changes
Not really sure. So far I've tested this out on a windows vm running some windows performance counter checks. I enabled this feature and saw that events were getting generated in accordance to the interval I set. I.E. if I set it to 5 min, events will be generated at 5 min intervals.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]
👍 for docs.

**Review by djmitche — CHANGES_REQUESTED** [review]
It looks like this only adds _documentation_ for the parameter.  I don't see an implementation for this parameter either in this PR or in the existing code.  Am I missing something?

**Comment by steveny91** [comment]
It's in the linked PR here:
https://github.com/DataDog/integrations-core/pull/12665

**Review by clarkb7 — CHANGES_REQUESTED** [review]
@steveny91 
> Sorry for the confusion! I'm not very familiar with this repo. Please do let me know if this needs to be added somewhere else as well.

the log has the line
```
Unknown key in config file: windows_counter_refresh_interval
```

Config options must be marked as known, see below for the different ways to do so
https://github.com/DataDog/datadog-agent/blob/7550de7079e1854f2fef7038c4f7b3ae2a73a14b/pkg/config/types.go#L75-L79

**Comment by clarkb7** [comment]
> Is there precedent for defining an agent config parameter like this, that's only used in integrations?

I don't know, but I would like to use it in the go checks as well https://github.com/DataDog/datadog-agent/commit/72b46c1ee07472a34a354f8b588c201c020b4f0b
https://github.com/DataDog/datadog-agent/blob/72b46c1ee07472a34a354f8b588c201c020b4f0b/pkg/util/winutil/pdhutil/pdhhelper.go#L79-L82

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -878,6 +878,16 @@ api_key:
     #
     # oid_batch_size: 5
 
+{{- if (eq .OS "windows")}}
+  ## @param windows_counter_refresh_interval - number - optional - default: 60
+  ## This controls the interval in seconds for which windows performance counters are refreshed by the agent.
+  ## Refreshes are necessary for the discovery of new counters/countersets and instances.
+  ## Refresh can be disabled by setting the interval to 0.
```
Comment (olivielpeau) [inline pkg/config/config_template.yaml:885]: This should be more explicit about the fact that it's used by Agent integrations, and ideally mention what integrations use this option.

**Review by olivielpeau — COMMENTED** [review]
It may make sense to also support this at the `init_config` level of integration configs, so that when Autodiscovery is used (in containerized environments), users can specify this directly as an integration config annotation/label instead of having to configure this at the agent level. But the precedence of the `init_config`-level option would need to be clearly defined and documented, as well as the limitation that if multiple integration configs define that option only one of the values will be picked up.

About this PR specifically: adding an option also needs an update to `pkg/config/config.go` to bind the env var option and (optionally) set a default value.

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (steveny91) [reply pkg/config/config_template.yaml]: Thanks for the feedback. I've added a better description. Please let me know if it needs to be even more detailed. 🙇

**Review by clarkb7 — APPROVED** [review]

**Review by djmitche — APPROVED** [review]

**Pull request merged**